sdk/src/androidTest/java/com/mapbox/maps/SettingsIntegrationTest.kt

```diff
@@ -0,0 +1,170 @@
+package com.mapbox.maps
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.mapbox.bindgen.Value
+import com.mapbox.common.SettingsServiceInterface
+import com.mapbox.common.SettingsServiceStorageType
+import com.mapbox.common.SettingsServiceFactory
+import com.mapbox.common.MapboxCommonSettings.LANGUAGE
+import com.mapbox.common.MapboxCommonSettings.WORLDVIEW
+import org.junit.*
+import org.junit.Assert.*
+import org.junit.experimental.runners.Enclosed
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Suite
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * SettingsService stores settings (eg. language and worldview) as {key:value} pair,
+ * there are no validation on keys and values on Settings interface, we perform tests
+ * based on set values to get the output from the data store used by settings service.
+ */
+@RunWith(Enclosed::class)
+@Suite.SuiteClasses(
+  SettingsIntegrationTest.SettingsInterfaceTest::class,
+  SettingsIntegrationTest.ParameterizedSettingsTest::class
+)
+class SettingsIntegrationTest {
+
+  @RunWith(AndroidJUnit4::class)
+  class SettingsInterfaceTest {
+    private lateinit var settingsServiceInterface: SettingsServiceInterface
+
+    @get:Rule
+    var rule = ActivityScenarioRule(EmptyActivity::class.java)
+
+    @Before
+    fun setUp() {
+      settingsServiceInterface =
+        SettingsServiceFactory.getInstance(SettingsServiceStorageType.PERSISTENT)
```

Comment:
should we check another `SettingsServiceStorageType`s?

Reply:
We should use `NON_PERSISTENT` until the UI blocking issue is resolved.